core/include/trajectory.hh

```diff
@@ -126,15 +126,15 @@ namespace aidaTT
     trajectory operator=(const trajectory&);
 
     // =========== members =========================================
-    trackParameters     _referenceParameters;
-    IFittingAlgorithm*  _fittingAlgorithm;
-    IPropagation*       _propagation;
-    const IGeometry* const _geometry;
+    trackParameters     _referenceParameters{};
+    IFittingAlgorithm*  _fittingAlgorithm{};
+    IPropagation*       _propagation{};
+    const IGeometry* const _geometry{};
```

Comment:
Default initializing const members like this doesn't make any sense. _geometry needs to be initialized in the initializer list, because it cannot be changed after that. So this sets a bad example, and may actually hide bugs that the compiler would otherwise complain about.

Reply:
@andresailer you are right in this particular case - _geometry should never be 0. In general your statement is of course wrong. One could very well have a const ptr that is either set to a value on construction or 0 for the lifetime of the object. Just imagine that a trajectory class might either deal with a realistic detector geometry or fall back to vacuum (no geometry given on construction)...

Reply:
I can imagine that it would be better to have an vacuum geometry object instead of a nullptr. But I see your point, that this might be sensible in some cases.